eclair-core/src/main/scala/fr/acinq/eclair/transactions/Scripts.scala

```diff
@@ -18,27 +18,15 @@ package fr.acinq.eclair.transactions
 
 import fr.acinq.bitcoin.Crypto.{PublicKey, ripemd160}
 import fr.acinq.bitcoin.Script._
-import fr.acinq.bitcoin.{ByteVector32, LexicographicalOrdering, LockTimeThreshold, OP_0, OP_1, OP_1NEGATE, OP_2, OP_2DROP, OP_ADD, OP_CHECKLOCKTIMEVERIFY, OP_CHECKMULTISIG, OP_CHECKSEQUENCEVERIFY, OP_CHECKSIG, OP_DROP, OP_DUP, OP_ELSE, OP_ENDIF, OP_EQUAL, OP_EQUALVERIFY, OP_HASH160, OP_IF, OP_NOTIF, OP_PUSHDATA, OP_SIZE, OP_SWAP, Satoshi, Script, ScriptElt, ScriptWitness, Transaction, TxIn}
+import fr.acinq.bitcoin.{ByteVector32, ByteVector64, Crypto, LexicographicalOrdering, LockTimeThreshold, OP_0, OP_1, OP_1NEGATE, OP_2, OP_2DROP, OP_ADD, OP_CHECKLOCKTIMEVERIFY, OP_CHECKMULTISIG, OP_CHECKSEQUENCEVERIFY, OP_CHECKSIG, OP_DROP, OP_DUP, OP_ELSE, OP_ENDIF, OP_EQUAL, OP_EQUALVERIFY, OP_HASH160, OP_IF, OP_NOTIF, OP_PUSHDATA, OP_SIZE, OP_SWAP, Satoshi, Script, ScriptElt, ScriptWitness, Transaction, TxIn}
 import scodec.bits.ByteVector
 
 /**
   * Created by PM on 02/12/2016.
   */
 object Scripts {
 
-  def toSelfDelay2csv(in: Int): Long = ???
-
-  /*in match {
-     case locktime(Blocks(blocks)) => blocks
-     case locktime(Seconds(seconds)) => TxIn.SEQUENCE_LOCKTIME_TYPE_FLAG | (seconds >> TxIn.SEQUENCE_LOCKTIME_GRANULARITY)
-   }*/
-
-  def expiry2cltv(in: Long): Long = ???
-
-  /*in match {
-      case locktime(Blocks(blocks)) => blocks
-      case locktime(Seconds(seconds)) => seconds
-    }*/
+  def der(sig: ByteVector64): ByteVector = Crypto.compact2der(sig) :+ 1
 
   def multiSig2of2(pubkey1: PublicKey, pubkey2: PublicKey): Seq[ScriptElt] = if (LexicographicalOrdering.isLessThan(pubkey1.toBin, pubkey2.toBin))
```

Comment:
Does it respect the spec ordering, defined as `the numerically lesser of the two DER-encoded`?

Reply:
Yes because we use lexicographical ordering for pubkeys (and their encoding does not change here) but not for signatures.

Reply:
More precisely: we still use DER encoding for public keys (the 33 bytes compressed representation of a public is DER)